queue_job_batch/models/queue_job_batch.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2019 Creu Blanca
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models, fields, api
+from odoo.addons.queue_job.job import job
+
+
+class QueueJobBatch(models.Model):
+    _name = 'queue.job.batch'
+    _inherit = ['mail.thread', 'mail.activity.mixin']
+    _description = 'Batch of jobs'
+    _log_access = False
+
+    name = fields.Char(
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    job_ids = fields.One2many(
+        'queue.job',
+        inverse_name='job_batch_id',
+        readonly=True,
+    )
+    job_count = fields.Integer(
+        compute='_compute_job_count',
+    )
+    user_id = fields.Many2one(
+        'res.users',
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('enqueued', 'Enqueued'),
+            ('progress', 'In Progress'),
+            ('finished', 'Finished')
+        ],
+        default='draft',
+        required=True,
+        readonly=True,
+        track_visibility='onchange',
+    )
+    finished_job_count = fields.Float(
+        compute='_compute_job_count',
+    )
+    failed_job_count = fields.Float(
+        compute='_compute_job_count',
+    )
+    company_id = fields.Many2one(
+        'res.company',
+        readonly=True,
+    )
+    is_read = fields.Boolean(
+        default=True
+    )
+    completeness = fields.Float(
+        compute='_compute_job_count',
+    )
+    failed_percentage = fields.Float(
+        compute='_compute_job_count',
+    )
+
+    def enqueue(self):
+        self.filtered(lambda r: r.state == 'draft').write({
+            'state': 'enqueued'
+        })
```

Comment:
You probably need to call `check_state` here, in case all jobs are already done when you enqueue the batch.

Reply:
You are right. Change applied